src/status_im/wallet/core.cljs

```diff
@@ -521,24 +544,45 @@
              (assoc-in [:wallet :request-transaction :amount-text] amount)
              (assoc-in [:wallet :request-transaction :amount-error] error))}))
 
-;;TODO request isn't implemented
 (fx/defn set-symbol-request
   {:events [:wallet.request/set-symbol]}
   [{:keys [db]} symbol]
   {:db (assoc-in db [:wallet :request-transaction :symbol] symbol)})
 
+(re-frame/reg-fx
+ ::resolve-address
+ (fn [{:keys [registry ens-name cb]}]
+   (ens/get-addr registry ens-name cb)))
+
+(fx/defn on-recipient-address-resolved
+  {:events [::recipient-address-resolved]}
+  [{:keys [db]} address]
+  {:db (assoc-in db [:wallet/prepare-transaction :to :address] address)})
+
 (fx/defn prepare-transaction-from-chat
   {:events [:wallet/prepare-transaction-from-chat]}
   [{:keys [db]}]
-  (let [identity (:current-chat-id db)]
-    {:db (assoc db :wallet/prepare-transaction
-                {:from (ethereum/get-default-account (:multiaccount/accounts db))
-                 :to   (or (get-in db [:contacts/contacts identity])
-                           (-> identity
-                               contact.db/public-key->new-contact
-                               contact.db/enrich-contact))
-                 :symbol :ETH
-                 :from-chat? true})}))
+  (let [chain (ethereum/chain-keyword db)
+        identity (:current-chat-id db)
+        {:keys [ens-verified name] :as contact}
+        (or (get-in db [:contacts/contacts identity])
+            (-> identity
+                contact.db/public-key->new-contact
+                contact.db/enrich-contact))]
+    (cond-> {:db (assoc db
+                        :wallet/prepare-transaction
+                        {:from (ethereum/get-default-account
+                                (:multiaccount/accounts db))
+                         :to contact
+                         :symbol :ETH
+                         :from-chat? true})}
+      ens-verified
+      (assoc ::resolve-address
```

Comment:
it might take some time, and also where might be an error ,so we need to handle this

Reply:
For now it is handled the same way as it is handled in the wallet